lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -670,6 +670,564 @@ class AuthList
+      $("#queryTableBody tr").remove()
+      queries = data['queries']
+      # use _.each instead of 'for' so that member
+      # is bound in the button callback.
+      _.each queries, (query) =>
+        query_id = query['id']
+        block_id = query['block_id']
+        block_type = query['block_type']
+        state_key = block_type + "/" + block_id
+        display_name = query['display_name']
+        display_entity = {'text':display_name, 'id':state_key}
+        filter_on = {'text':query['filter_on']}
+        inclusion = {'text':query['inclusion']}
+        done = query['done']
+        type = {'text':query['type']}
+        arr = [inclusion, type, display_entity, filter_on, done]
+        @tr = @start_row(inclusion['text'].toLowerCase(),arr,
+          {'class':["working"],'query':query_id},  $("#queryTableBody"))
+        @check_done()
+
+
+  get_saved_queries: (cb)->
+    $.ajax
+      dataType: 'json'
+      url: $("#savedQueriesTable").data 'endpoint'
+      data: rolename: 'instructor'
+      success: (data) -> cb? null, data
+      error: std_ajax_err ->
+        cb? gettext("Error getting saved queries")
+
+  #get a user's saved queries and load them into saved queries
+  load_saved_queries: ->
+    $("#savedQueriesTable tr").remove()
+    $("#invisibleQueriesStorage tr").remove()
+    @get_saved_queries (error, data) =>
+      # abort on error
+      return @show_errors error unless error is null
+      queries = data['queries']
+      groups = new Set()
+      _.each queries, (query) =>
+        block_id = query['block_id']
+        block_type = query['block_type']
+        state_key = block_type + "/" + block_id
+        display_name = query['display_name']
+        display_entity = {'text':display_name, 'id':state_key}
+        filter_on = {'text':query['filter_on']}
+        inclusion = {'text':query['inclusion']}
+        created = query['created']
+        type = {'text':query['type']}
+        arr = [inclusion,type, display_entity, filter_on]
+        invisibleTable = $("#invisibleQueriesStorage")
+        @tr = @start_row(inclusion['text'],arr,
+          {'class':["saved" + query.group]}, invisibleTable)
+        @tr[0].setAttribute('created',created)
+        groups.add(query.group)
+      group_arr = []
+      iter = groups.values()
+      val = iter.next()
+      while (val['done'] == false)
+        group_arr.push(val['value'])
+        val = iter.next()
+      group_arr.sort((a, b) ->return b-a)
+      for group in group_arr
+        lookup = ".saved" + group
+        saved_qs = $(lookup)
+        types = []
+        names = []
+        time = ""
+        for query in saved_qs
+          cells = query.children
+          types.push(cells[0].innerText)
+          names.push(cells[2].innerText)
+          time = query.getAttribute('created')
+        display_st = ""
+        for i in [0..types.length-1]
+          display_st += types[i]
+          display_st += " "
+          display_st += names[i] + " "
```

Comment:
This was the intention. If there is no trailing space it would look like
"NOT IntroductionAND Introduction"  vs  "NOT Introduction AND Introduction "
I didn't want to create a separate array and then join it with spaces